src/common/heartbeat/src/producer.cpp

```diff
@@ -1,28 +1,62 @@
 #include <rclcpp/rclcpp.hpp>
-#include <std_msgs/msg/bool.hpp>
+#include <builtin_interfaces/msg/time.hpp>
+#include <common_interfaces/srv/handshake.hpp>
 
+#include "heartbeat/date.h"
+
+#include <chrono>
 #include <iostream>
+#include <sstream>
+#include <random>
 
 class Producer : public rclcpp::Node {
 public:
     Producer() : Node("heartbeat_producer") {
         using namespace std::chrono_literals;
-        heartbeat_publisher = this->create_publisher<std_msgs::msg::Bool>("heartbeat", 10);
-        heartbeat_timer = this->create_wall_timer(1s, std::bind(&Producer::heartbeat, this));
+        heartbeat_service_listener = this->create_service<common_interfaces::srv::Handshake>("heartbeat_handshake", std::bind(&Producer::handshake, this, std::placeholders::_1, std::placeholders::_2));
+        heartbeat_pub = this->create_publisher<builtin_interfaces::msg::Time>("heartbeat", 10);
+        srand(time(NULL));
     }
 private:
+    void handshake(const common_interfaces::srv::Handshake::Request::SharedPtr req,
+            common_interfaces::srv::Handshake::Response::SharedPtr res) {
+        using namespace date; // i use date library to make outputting chrono::time_point not hell (not necessary in c++20)
```

Comment:
You repeat this line a lot, could you just stick it at the top of the file rather than scoping it in?

While you're at it you could `using namespace std::chrono` too :grin: